src/GraphQl/Executor.php

```diff
@@ -24,11 +25,19 @@
  */
 final class Executor implements ExecutorInterface
 {
+    public function __construct(private readonly bool $graphQlIntrospectionEnabled = true)
+    {
+    }
+
     /**
      * {@inheritdoc}
      */
     public function executeQuery(Schema $schema, $source, mixed $rootValue = null, mixed $context = null, array $variableValues = null, string $operationName = null, callable $fieldResolver = null, array $validationRules = null): ExecutionResult
     {
+        $validationRules[] = new DisableIntrospection(
```

Comment:
You need to handle the case where `$validationRules` is `null`.

Reply:
I updated the code to instantiate the rule globally for both reasons:
1) it will not create a new rule (`new DisableIntrospection()`) on each graphql query
2) the configuration yaml-file setup the bundle globally

 ... but  the code is now in the constructor.

@alanpoulain are you ok with that or I keep the previous logic ?